vsintegration/src/FSharp.Editor/Commands/DumpCommandService.fs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.ComponentModel.Composition
+open System.Diagnostics
+open Microsoft.VisualStudio
+open Microsoft.VisualStudio.Text.Editor
+open Microsoft.VisualStudio.Utilities
+open Microsoft.VisualStudio.Shell
+open Microsoft.VisualStudio.Shell.Interop
+open Microsoft.VisualStudio.FSharp.Interactive
+open System.ComponentModel.Design
+open Microsoft.Diagnostics.Runtime
+
+
+[<Export(typeof<IWpfTextViewCreationListener>)>]
+[<ContentType(FSharpConstants.FSharpContentTypeName)>]
+[<TextViewRole(PredefinedTextViewRoles.PrimaryDocument)>]
+type internal DumpCommandFilterProvider 
+    [<ImportingConstructor>] 
+    (checkerProvider: FSharpCheckerProvider,
+     [<Import(typeof<SVsServiceProvider>)>] serviceProvider: System.IServiceProvider) =
+
+    let projectSystemPackage =
+      lazy(
+        let shell = serviceProvider.GetService(typeof<SVsShell>) :?> IVsShell
+        let packageToBeLoadedGuid = ref (Guid "{91a04a73-4f2c-4e7c-ad38-c1a68e7da05c}")  // FSharp ProjectSystem guid        
+        match shell.LoadPackage packageToBeLoadedGuid with
+        | VSConstants.S_OK, pkg ->
+            pkg :?> Package
+        | _ -> null)
+
+    let FSharpDump (_this:Package) (_sender:obj) (_e:EventArgs) =
+        checkerProvider.Checker.ClearLanguageServiceRootCachesAndCollectAndFinalizeAllTransients()
+        use target = DataTarget.CreateSnapshotAndAttach(Process.GetCurrentProcess().Id)
+        let runtime = target.ClrVersions.[0].CreateRuntime();
+        let outputPane = projectSystemPackage.Value.GetOutputPane(VSConstants.OutputWindowPaneGuid.BuildOutputPane_guid, "FSharp Dump")
```

Comment:
I'm not familiar with this API.  Does "FSharp Dump" need to be localized/translated?  If so it'll have to be added to a `.resx` file and consumed that way.